cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -140,17 +188,27 @@ public void getReports(HttpServiceRequest request, HttpServiceResponder responde
       responder.sendJson(200, reportList);
     }
 
+    private String getUserName(Map<String, List<String>> headers) {
+      return headers.getOrDefault(USER_ID, Collections.emptyList()).stream().findFirst().orElse(DEFAULT_USER_ID);
```

Comment:
Minor optimization - if you replace the `emptyList()` with `singletonList(DEFAULT_USER_ID)` then you can remove the `orElse()` part at the end.

Reply:
good point, i have updated it.